### PR TITLE
[Features]Support dump segment predition

### PR DESCRIPTION
**Pull request by MeowZheng**
## Motivation

1. It is used to save the segmentation predictions as files and upload these files to a test server

## Modification

1. Add output_file and format only in `IoUMetric`
 
## BC-breaking (Optional)

No

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
3. The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
4. If the modification has potential influence on downstream projects, this PR should be tested with downstream projects, like MMDet or MMDet3D.
5. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

File: docs/en/user_guides/4_train_test.md [hunk 1]
```diff
@@ -218,3 +218,92 @@ You can check [the source code](../../../tools/slurm_test.sh) to review full arg
...
+
+```shell
+python tools/test.py ${CONFIG_FILE} ${CHECKPOINT_FILE} --cfg-options test_evaluator.output_dir=${OUTPUT_DIR}
+```
+
+Here is an example to save the predicted results from model `fcn_r50-d8_4xb4-80k_ade20k-512x512` on ADE20k validatation dataset.
+
+```shell
+python tools/test.py configs/fcn/fcn_r50-d8_4xb4-80k_ade20k-512x512.py ckpt/fcn_r50-d8_512x512_80k_ade20k_20200614_144016-f8ac5082.pth --cfg-options test_evaluator.output_dir=work_dirs/format_results
+```
```
Comment (xiexinch) [inline docs/en/user_guides/4_train_test.md:236]: Should we add an argument to the `test.py` script like mmyolo?
https://github.com/open-mmlab/mmyolo/blob/main/tools/test.py#L28

**Comment by AI-Tianlong** [comment]
Excellent work!!!! I really need this feature, hope to merge soon😍

**Review by csatsurnh — APPROVED** [review]

File: docs/en/user_guides/4_train_test.md [hunk 2]
```diff
@@ -218,3 +218,98 @@ You can check [the source code](../../../tools/slurm_test.sh) to review full arg
 CUDA_VISIBLE_DEVICES=0,1,2,3 GPUS=4 MASTER_PORT=29500 sh tools/slurm_train.sh ${PARTITION} ${JOB_NAME} config1.py ${WORK_DIR}
 CUDA_VISIBLE_DEVICES=4,5,6,7 GPUS=4 MASTER_PORT=29501 sh tools/slurm_train.sh ${PARTITION} ${JOB_NAME} config2.py ${WORK_DIR}
 ```
+
+## Testing and saving predicted segment files
```
Comment (xiexinch) [inline docs/en/user_guides/4_train_test.md:222]:
```suggestion
## Testing and saving segmentation results
```

File: docs/zh_cn/user_guides/4_train_test.md [hunk 3]
```diff
@@ -223,3 +223,95 @@ GPUS=4 sh tools/slurm_train.sh dev pspnet configs/pspnet/pspnet_r50-d8_512x1024_
 CUDA_VISIBLE_DEVICES=0,1,2,3 GPUS=4 MASTER_PORT=29500 sh tools/slurm_train.sh ${分区} ${任务名} config1.py ${工作路径}
 CUDA_VISIBLE_DEVICES=4,5,6,7 GPUS=4 MASTER_PORT=29501 sh tools/slurm_train.sh ${分区} ${任务名} config2.py ${工作路径}
 ```
+
+## 测试并保存输出分割结果
```
Comment (xiexinch) [inline docs/zh_cn/user_guides/4_train_test.md:227]:
```suggestion
## 测试并保存分割结果
```

**Review by xiexinch — APPROVED** [review]

**Pull request merged**